cmd/thanos/compact.go

```diff
@@ -223,6 +235,60 @@ func runCompact(
 		}, func(error) {
 			cancel()
 		})
+
+		g.Add(func() error {
```

Comment:
Some comment would be nice here

Reply:
Also I would vote to actually get it out of separate run.Group and put compaction, downsampling and retention to the SAME run group.

There reason is strong: run.Group are made for concurrency. And there is no concurrency because we need to add mutexes (and it make sense - we don't want to run both retention and compaction/downsampling). If we don't care about concurency we should actually have one run.Group for all 3 things - we can be sure that all things will be done properly. It is just easier to maintain. Think about this: 
- What if `f` function will be running over 5 minutes (which is MOSTLY the case) and the `Repeat` will immidiately run `f` again and gain mutex again. This will mean that there is no guarantee the retention will have time slot to run (: so.. potential resource starvation.

Reply:
But we need to make sure to spread all into ~3 smaller function to make it clear